service/history/workflow/workflow_task_state_machine.go

```diff
@@ -867,6 +875,13 @@ func (m *workflowTaskStateMachine) AddWorkflowTaskTimedOutEvent(
 			workflowTask.StartedEventID,
 			enumspb.TIMEOUT_TYPE_START_TO_CLOSE,
 		)
+
+		if m.ms.executionInfo.LastWorkflowTaskFailure == nil {
+			m.ms.executionInfo.LastWorkflowTaskFailure = &persistencespb.WorkflowExecutionInfo_LastWorkflowTaskTimedOutType{
+				LastWorkflowTaskTimedOutType: enumspb.TIMEOUT_TYPE_SCHEDULE_TO_CLOSE,
```

Comment:
Above we set this to START_TO_CLOSE, I would extract that value to a variable and set in both places.